Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -600,62 +307,202 @@ namespace gdjs {
+      this._downKey ||
+        (this._downKey =
+          !this._ignoreDefaultControls && inputManager.isKeyPressed(DOWNKEY));
+
+      this._releaseKey ||
+        (this._releaseKey =
+          !this._ignoreDefaultControls && inputManager.isKeyPressed(DOWNKEY));
+
+      this._requestedDeltaX += this._updateSpeed(timeDelta);
+
+      //0.2) Track changes in object size
+      this._state.beforeUpdatingObstacles();
+      this._onFloor._oldHeight = object.getHeight();
+
+      //0.3) Update list of platforms around/related to the object
+
+      //Compute the list of the objects that will be used
+      this._updatePotentialCollidingObjects(
+        Math.max(this._requestedDeltaX, this._maxFallingSpeed * timeDelta)
+      );
+      this._updateOverlappedJumpThru();
+
+      //1) X axis:
+      this._state.checkTransitionBeforeX();
+      this._state.beforeMovingX();
+
+      //Ensure the object is not stuck
+      if (this._separateFromPlatforms(this._potentialCollidingObjects, true)) {
+        this._canJump = true;
+      }
+      //After being unstuck, the object must be able to jump again.
+
+      const oldX = object.getX();
+      this._moveX();
+
+      //2) Y axis:
+      this._state.checkTransitionBeforeY(timeDelta);
+      this._state.beforeMovingY(timeDelta, oldX);
 
-          //Ensure nothing is grabbed.
-          this._releaseGrabbedPlatform();
-        } else {
-          // Avoid landing on a platform if the object is not going down.
-          // (which could happen for Jumpthru, when the object jump and pass just at the top
-          // of a jumpthru, it could be considered as landing if not this this extra check).
-          const canLand = requestedDeltaY >= 0;
-
-          //Check if landing on a new floor: (Exclude already overlapped jump truh)
-          let collidingPlatform = this._getCollidingPlatform();
-          if (canLand && collidingPlatform !== null) {
-            this._isOnFloor = true;
-            this._canJump = true;
-            this._jumping = false;
-            this._currentJumpSpeed = 0;
-            this._currentFallSpeed = 0;
-
-            //Register the colliding platform as the floor.
-            this._floorPlatform = collidingPlatform;
-            this._floorLastX = this._floorPlatform.owner.getX();
-            this._floorLastY = this._floorPlatform.owner.getY();
-
-            //Ensure nothing is grabbed.
-            this._releaseGrabbedPlatform();
-          } else {
-            //In the air
-            this._canJump = false;
-            this._isOnFloor = false;
-            this._floorPlatform = null;
-          }
-        }
-        object.setY(oldY);
+      const oldY = object.getY();
+      this.moveY();
+
+      //3) Update the current floor data for the next tick:
+      this._updateOverlappedJumpThru();
+      //TODO what about a moving platforms, remove this condition to do the same as for grabbing?
```

Comment:
Yes probably! This means that for now a moving ladder will create issues - though I'm not sure exactly why.
I reckon we should leave it here for this PR, but do a PR with it removed and a few tests that verify how it works (probably we the character should follow the ladder, be pushed by any potential colliding obstacle and start falling if not on the ladder anymore).